services/pypi/pypi-typing.service.js

```diff
@@ -0,0 +1,40 @@
+import PypiBase, { pypiGeneralParams } from './pypi-base.js'
+
+export default class PypiTyping extends PypiBase {
+  static category = 'platform-support'
+
+  static route = this.buildRoute('pypi/typing')
+
+  static openApi = {
+    '/pypi/typing/{packageName}': {
+      get: {
+        summary: 'PyPI - Typing',
+        description:
+          'Whether the package provides type information (inline annotations or stub files)',
```

Comment:
This isn't quite true. What this badge is actually telling us is whether the package metadata contains the classifier `Typing :: Typed`. You can push a package to PyPI that does have annotations but does not use that classifier, and vice versa.

Reply:
I wasn't sure whether to focus more on the intent of the badge or the implementation details here.

"whether or not the package has types" is very complicated, even more so if I think about including standalone stubs packages, typeshed, etc. 

My plan was to have the badge first, and then follow up by making sure that packages have types all have this classifier in the metadata.

There's some context and discussion about the badge here: https://discuss.python.org/t/github-badge-for-type-coverage/74116/

Reply:
I was really just meaning the badge description should accurately explain what we're actually doing/showing.

It is outside the scope of what we can do on a badge to actually perform static analysis on a package in the context of a request/response cycle, but if there were a service that did this and exposed type coverage/whether the types for a package are inline or stubs/etc for a given package, we could certainly consider adding badges that consume that data in future. Similar to how we have PyPI downloads badges using the data from https://pypistats.org/